src/opm/parser/eclipse/EclipseState/Aquifetp.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+  Copyright (C) 2017 TNO
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <opm/parser/eclipse/EclipseState/Aquifetp.hpp>
+
+
+namespace Opm{
+	
+  Aquifetp::Aquifetp(const EclipseState& eclState, const Deck& deck)
+  	{	
+      if (!deck.hasKeyword("AQUFETP"))
+        return;
+
+      const auto& aquifetpKeyword = deck.getKeyword("AQUFETP");
+
+      for (auto& aquifetpRecord : aquifetpKeyword){
+
+          Aquifetp::AQUFETP_data data;
+
+          data.aquiferID = aquifetpRecord.getItem("AQUIFER_ID").template get<int>(0);
```

Comment:
Quite few of the items in the `AQUFETP` keyword have a defined default; the call:
```
 aquifetpRecord.getItem("P0").getSIDouble(0);
```
will blow up with an exception if the`P0` item has not been given a value in the input deck. In cases like this you must protect like this:
```
const auto& item = record.getItem("P0"):
if (item.hasValue(0))
    data.p0 = item.getSIDouble(0);
```